### PR TITLE
Documentation via sphinx

### PR DESCRIPTION
**Pull request by Pike**
This is mostly a port of what python-fluent does. I actually just copied the doc trees and then edited it to work.

The first commit is making the doc strings useful.

I then manually copied them over into the reference documentation, just because I think that dokka output is so sub-par.

One thing that I wish would work, but doesn't, it using Kotlin syntax highlighting for the included `Nodes.kt` for the AST reference. Better than nothing, still sad. It's claiming it couldn't highlight that, but it can. At least pygments commandline UI can. No idea.

### CODE REVIEW

File: fluent.syntax/src/main/kotlin/org/projectfluent/syntax/parser/Parser.kt [hunk 1]
```diff
@@ -727,7 +732,7 @@ class FluentParser(var withSpans: Boolean = false) {
     }
 }
 
-data class Indent(var value: String) : PatternElement() {
+internal data class Indent(var value: String) : PatternElement() {
```
Comment (stasm) [inline fluent.syntax/src/main/kotlin/org/projectfluent/syntax/parser/Parser.kt:735]: `private`?

File: fluent.syntax/src/main/kotlin/org/projectfluent/syntax/parser/Parser.kt [hunk 2]
```diff
@@ -2,9 +2,14 @@ package org.projectfluent.syntax.parser
 
 import org.projectfluent.syntax.ast.*
 
-var trailingWSRe = Regex("[ \t\n\r]+\$")
-val VALID_FUNCTION_NAME = Regex("^[A-Z][A-Z0-9_-]*\$")
-
+internal val trailingWSRe = Regex("[ \t\n\r]+\$")
+internal val VALID_FUNCTION_NAME = Regex("^[A-Z][A-Z0-9_-]*\$")
```
Comment (stasm) [inline fluent.syntax/src/main/kotlin/org/projectfluent/syntax/parser/Parser.kt:6]: Should both of these be `private` instead? They're only used in this file.

File: docs/theme/layout.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html lang="en" data-url_root="{{ pathto('', 1) }}">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
```
Comment (stasm) [inline docs/theme/layout.html:6]: Does the title update automatically?

File: fluent.syntax/docs/parsing.rst [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+   /**
+    * Parse Fluent resources.
+    *
+    * @property withSpans create source positions for nodes (not supported yet)
+    */
+   class FluentParser(var withSpans: Boolean = false) {
+
+       fun parse(source: String): Resource {
+       }
```
Comment (stasm) [inline fluent.syntax/docs/parsing.rst:18]: I think you can omit the braces around the empty method body:

```kotlin
class FluentParser(var withSpans: Boolean = false) {
   fun parse(source: String): Resource
}
```

Comment (stasm) [reply fluent.syntax/docs/parsing.rst:18]: Also note that there's a missing closing brace at the end of this block.

File: docs/theme/layout.html (hunk 3 above)
Comment (Pike) [reply docs/theme/layout.html:6]: No. Also, the complete `theme` directory is from an experimental branch that I accidentally started with, and then didn't fully remove. Gonna remove that.

**Pull request merged**